### PR TITLE
Fix the regular user calling the iasql_ funcs and demonstrate a nicer install

### PR DESCRIPTION
**Pull request by dfellis**
Part of #538 but not complete

### CODE REVIEW

**Comment by dfellis** [comment]
Improved error messaging:
Screenshot from 2022-03-23 20-36-43

Success path returning useful metadata as a virtu
Screenshot from 2022-03-23 20-37-22
al table:

File: src/migration/1647526647810-init.ts [hunk 1]
```diff
@@ -99,11 +99,22 @@ export class init1647526647810 implements MigrationInterface {
...
+              record_count int
+            )
             language plpgsql
             as $$
             begin
                 perform until_iasql_operation('INSTALL', _mods);
+                return query select
+                  m.name as module_name,
+                  t.table as table_name,
+                  (xpath('/row/c/text()', query_to_xml(format('select count(*) as c from public.%I', t.table), FALSE, TRUE, '')))[1]::text::int AS record_count
```
Comment (depombo) [inline src/migration/1647526647810-init.ts:114]: what is this line doing? looking at the logs in a file?

File: src/services/db-manager.ts [hunk 2]
```diff
@@ -74,6 +74,8 @@ export function newPostgresRoleQuery(user: string, pass: string, dbId: string) {
     GRANT USAGE, SELECT ON ALL SEQUENCES IN SCHEMA graphile_worker TO ${user};
     CREATE POLICY ${user}_sel_job ON graphile_worker.jobs FOR SELECT TO ${user} USING (true);
     CREATE POLICY ${user}_ins_job ON graphile_worker.jobs FOR INSERT TO ${user} WITH CHECK (true);
+    CREATE USER MAPPING IF NOT EXISTS FOR ${user} SERVER loopback_dblink_${dbId} OPTIONS (user '${config.dbUser}', password '${config.dbPassword}');
+    GRANT USAGE ON FOREIGN SERVER loopback_dblink_${dbId} TO ${user};
```
Comment (depombo) [inline src/services/db-manager.ts:78]: noice

**Review by depombo — APPROVED** [review]

File: src/migration/1647526647810-init.ts (hunk 1 above)
Comment (dfellis) [reply src/migration/1647526647810-init.ts:114]: I found the pattern here: https://stackoverflow.com/a/2611745

It's a trick to be able to do a query where the table name itself is dynamic by abusing Postgres' XML functions, similar in spirit to your "DB connects to itself as a foreign database" trick.

**Review by depombo — APPROVED** [review]

**Comment by dfellis** [comment]
Absolutely pointless but lol
Screenshot from 2022-03-23 21-00-53

**Pull request merged**